driver/src/driver/stablex_driver.rs

```diff
@@ -14,35 +15,61 @@ pub struct StableXDriver<'a> {
     past_auctions: HashSet<U256>,
     contract: &'a dyn StableXContract,
     price_finder: &'a mut dyn PriceFinding,
+    metrics: StableXMetrics,
 }
 
 impl<'a> StableXDriver<'a> {
-    pub fn new(contract: &'a dyn StableXContract, price_finder: &'a mut dyn PriceFinding) -> Self {
+    pub fn new(
+        contract: &'a dyn StableXContract,
```

Comment:
I never know when to use these `&'a` things. IS this something clippy tells you, or do you just know.

Reply:
It's the lifetime of the reference you are passing. You need to make sure that the person "owning" the object doesn't get rid of it before the person "borrowing" it goes out of scope (in other languages you might have a dangling pointer otherwise, which the rust compiler doesn't allow you to have).
It can be omitted if the compiler can automatically infer and thus verify the lifetimes are ok.

https://doc.rust-lang.org/1.9.0/book/lifetimes.html has a more detailed explanation.